src/libraries/System.Private.CoreLib/src/System/Int128.cs

```diff
@@ -742,12 +742,19 @@ public static (Int128 Quotient, Int128 Remainder) DivRem(Int128 left, Int128 rig
 
         /// <inheritdoc cref="IBinaryInteger{TSelf}.LeadingZeroCount(TSelf)" />
         public static Int128 LeadingZeroCount(Int128 value)
+            => value.LeadingZeroCount();
+
+        /// <summary>Computes the number of leading zero bits in this value.</summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private int LeadingZeroCount()
         {
+            Int128 value = this;
```

Comment:
Why do we need to make a copy of `this`, rather than just reading `_lower` and `_upper` directly?

Reply:
and why does it need to be an instance method at all?

Reply:
> Why do we need to make a copy of this, rather than just reading _lower and _upper directly?

I did that to ensure there's a consistent view of the data, so that even if there's a tear while reading the value, subsequent operations on it will be consistent, e.g. the check on _upper and then the use of _upper.  If we don't think that's important, I can remove the local.

> and why does it need to be an instance method at all?

Because you can't overload by return type, so we'd need to name it something different if it's taking the same arguments as the existing method.  Is that what you'd prefer?

Reply:
I think having a different name, like `LeadingZeroCountCore` would be fine here. That allows it to be `static`, allows the minor optimization of not needing the JIT to promote and remove `upper` since it's unused, etc.

It being instance and forcing the value to become address taken and then additionally inserting a copy feels like its create more work and is likely to de-optimize things in the long run.